File: go-client-sdk/pkg/models/operations/getinvoiceinvoiceid.go
package operations

import (
	"github.com/speakeasy-sdks/orb-sdks/go-client-sdk/pkg/models/shared"
)

type GetInvoiceInvoiceIDPathParams struct {
	InvoiceID string `pathParam:"style=simple,explode=false,name=invoice_id"`
}

type GetInvoiceInvoiceIDRequest struct {
	PathParams GetInvoiceInvoiceIDPathParams
}

type GetInvoiceInvoiceIDResponse struct {
	ContentType string
	Invoice     *shared.Invoice
	StatusCode  int64
}

File: typescript-client-sdk/src/sdk/models/operations/getinvoiceinvoiceid.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class GetInvoiceInvoiceIdPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=invoice_id" })
  invoiceId: string;
}


export class GetInvoiceInvoiceIdRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: GetInvoiceInvoiceIdPathParams;
}


export class GetInvoiceInvoiceIdResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  invoice?: shared.Invoice;

  @SpeakeasyMetadata()
  statusCode: number;
}
